doc/source/development/rfc/rfc104_gdal_cli.rst

```diff
@@ -0,0 +1,1175 @@
+      --no-md                                              Suppress metadata printing
+      --no-ct                                              Suppress color table printing
+      --no-fl                                              Suppress file list printing
+      --checksum                                           Compute pixel checksum
+      --list-mdd                                           List all metadata domains available for the dataset
+      --mdd <MDD>                                          Report metadata for the specified domain. 'all' can be used to report metadata in all domains
+
+    Esoteric Options:
+      --no-nodata                                          Suppress retrieving nodata value
+      --no-mask                                            Suppress mask band information
+      --subdataset <SUBDATASET>                            Use subdataset of specified index (starting at 1), instead of the source dataset itself
+
+
+A few invocations of "gdal raster info [OPTIONS] <FILENAME>"
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  .. code-block:: shell
+
+    $ gdal raster info byte.tif
+    [ ... JSON output stripped ... ]
+
+    $ gdal raster info -i byte.tif
+    [ ... JSON output stripped ... ]
+
+    $ gdal raster info --input byte.tif
+    [ ... JSON output stripped ... ]
+
+    $ gdal raster info --input=byte.tif
+    [ ... JSON output stripped ... ]
+
+    $ gdal raster info byte.tif --stats --format=text
+    [ ... text output stripped ... ]
+
+
+Using just ``gdal info <FILENAME>``
+
+  .. code-block:: shell
+
+    $ gdal raster byte.tif
+    [ ... JSON output stripped ... ]
+
+
+And cherry-on-the-cake ``gdal <FILENAME>``
+
+  .. code-block:: shell
+
+    $ gdal byte.tif
+    [ ... JSON output stripped ... ]
+
+
+"gdal [info] <FILENAME>" on dataset with mixed raster and vector content
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  .. code-block:: shell
+
+    $ gdal info mixed.gpkg
+    ERROR 1: 'mixed.gpkg' has both raster and vector content. Please use 'gdal raster info' or 'gdal vector info'.
+
+    $ gdal mixed.gpkg
+    ERROR 1: 'mixed.gpkg' has both raster and vector content. Please use 'gdal raster info' or 'gdal vector info'.
+
+
+A few invocations of "gdal raster convert"
+++++++++++++++++++++++++++++++++++++++++++
+
+  .. code-block:: shell
+
+    $ gdal raster convert byte.tif out.tif
+
+    $ gdal raster convert byte.tif out.tif --co=TILED=YES,COMPRESS=LZW
+    ERROR 1: File 'out.tif' already exists. Specify the --overwrite option to overwrite it.
+
+    $ gdal raster convert --input=byte.tif --output=out.tif --co=TILED=YES,COMPRESS=LZW --overwrite
+
+    $ gdal raster convert -i byte.tif -o out.tif --co=TILED=YES,COMPRESS=LZW --overwrite --progress
+    0...10...20...30...40...50...60...70...80...90...100 - done.
+
+
+"raster" has a declared short alias of "r" and "convert" of "c", so you can also do:
```

Comment:
In my personal opinion, those alias `r` or `c` make the syntax very difficult to understand. And misleading. `raster`, `read`, what was `r` for?
I prefer not having that option.

Reply:
Was a @jratike80 suggestion in https://github.com/OSGeo/gdal/pull/11216#issuecomment-2464924674, but like you, I'm not too convinced

Reply:
Well, I was thinking that people tend to be lazy when they type, and because of that they will write `gdal info my_geopackage.gpkg`, and then sometimes be adviced to rerun explicitly with either `raster` or `vector`. Maybe a short form could tempt them to be explicit from the beginning. First I thought it could be `gdal rinfo`, `gdal vinfo` and `gdal mdinfo` but it did not feel good for all the other subcommands. Anyway, it is not hard for me to give up this idea.

Reply:
short aliases suppressed